scripts/dockerrun.sh

```diff
@@ -87,8 +87,43 @@ fi
 #  VSO
 [ ! -z "$BUILD_BUILDID" ] && DOTNET_BUILD_CONTAINER_NAME="${BUILD_BUILDID}-${BUILD_BUILDNUMBER}"
 
+# Executes a command and retries if it fails.
+# NOTE: This function is the exact copy from init-docker.sh.
+# Reason for not invoking init.docker.sh directly is since that script 
+# also performs cleanup, which we do not want in this case.
```

Comment:
To me that seems like a reason to split that part of `init-docker.sh` into, say, `pull-image.sh`, instead. @MichaelSimons